File: Python/algorithmic_toolbox/04_fibonacci_last_digit.py
import sys

MODULUS = 10

def fibonacci_last_digit(n):
    k = 1
    a, b = 0, 1
    while k <= n:
        a, b = b, (a + b) % MODULUS
        k += 1
    return a


def main():
    data = sys.stdin.read().splitlines()
    n = int(data[0])
    digit = fibonacci_last_digit(n)
    print(digit)


if __name__ == "__main__":
    main()

File: R/algorithmic_toolbox/04_fibonacci_last_digit.R

modulus <- 10

FibonacciLastDigit <- function(n) {
    k <- 1
    a <- 0
    b <- 1
    while (k <= n) {
        c <- (a + b) %% modulus
        a <- b
        b <- c
        k <- k + 1
    }
    a
}

Main <- function() {
    conn <- file("stdin", "r")
    n <- as.integer(readLines(conn, n=1))
    close(conn)
    digit <- FibonacciLastDigit(n)
    cat(digit, fill = TRUE)
}

if (!interactive()) {
    Main()
}
